### ISSUE TITLE
Faster Paging

### ISSUE THREAD

**Issue by djchapm** [awslabs/aws-c-s3#335]
### Describe the feature

Provide pagination metadata so that we may perform retrievals in parallel.  i.e. async listObjectsV2, we never know how many pages there are and are unable to pull a page until we know the page before it and a dynamic 'next-token'.

Or suggest strategies - thoughts:
- Ability to modify listObjectsV2Request.maxKeys to something > 1000.
- Push down a read-ahead parameter so subsequent page retrievals can happen before we actually request the next page.
- Ability to retrieve commonPrefix only when delimiter is set, ignoring the loading of actual keys.  This would allow us to get a handle on our key patterns and subsequently make parallel listObjectRequests where we find a match.
- Wildcard ability in the prefix.  We know the start(prefix) and some of the middle portions of the key due to our partition strategy.

### Use Case

highly complex system writing streaming data to S3 - on disaster recovery we need to scan S3 objects, identify key patterns and determine where we left off in order to restart at the correct location.  Across something like 10Million keys.  

Today max

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by TingDaoK** [comment]
We are not exposing the `listObjectsV2` API. I guess you are using Java SDK, and they have their own impl for the paging instead of using us. So, maybe cut the issue to them instead.

**Comment by yasminetalby** [comment]
Hello @djchapm ,

Thank you very much for your submission
If you confirm that you are using the JAVA SDK could you close this ticket and open a ticket to the AWS JAVA SDK V2 repository : https://github.com/aws/aws-sdk-java-v2/issues for tracking purposes.

Thank you very much for your time and collaboration.

Sincerely,

Yasmine

**Comment by djchapm** [comment]
Yes using AWS Java SDK V2 - created ticket https://github.com/aws/aws-sdk-java-v2/issues/4246.

Thanks!
